src/AppBundle/Controller/TodoController.php

```diff
@@ -0,0 +1,242 @@
+<?php
+/*
+ * Routes for accessing the To do info.
+ *
+ * All of these paths are behind the firewall for ROLE_USER.
+ *
+ * All of these paths consume and send application/json
+ */
+
+namespace AppBundle\Controller;
+
+use AppBundle\ControllerTemplates\JSONController;
+use AppBundle\ControllerTemplates\JSONControllerException;
+use AppBundle\Document\Todo;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\JsonResponse;
+
+class TodoController extends JSONController
+{
+    /**
+     * @Route("/todo", name="get_todo")
+     * @Method({"GET"})
+     *
+     * Gets the ids of all items.
+     *
+     * @return JsonResponse
+     */
+    public function getTodoAction()
+    {
+        //Get all to do for this user
+        $todos = $this->dm()->createQueryBuilder('AppBundle:Todo')
+            ->field('userId')->equals($this->getUser()->getIdString())
+            ->getQuery()->execute();
+
+        //get the _id, title, and completed
+
+        $output = array();
+        foreach ($todos as $todo) {
+            $temp = array();
+            $temp['_id'] = '' . $todo->getId();
+            $temp['completed'] = $todo->getCompleted();
+            $temp['title'] = $todo->getTitle();
+            array_push($output, $temp);
+        }
+
+        return new JsonResponse($output);
+    }
+
+    /**
+     * @Route("/todo", name="delete_todo")
+     * @Method({"DELETE"})
+     *
+     * Deletes all of the existing To do entries for this user.
+     *
+     * @return JsonResponse
+     */
+    public function deleteTodoAction()
+    {
+        $itemsDeleted = 0;
+        try {
+            $collection = $this->dm()->getDocumentCollection('AppBundle:Todo');
+            //The result from the mongodb operation
+            $mongoResult = $collection->remove(array('userId' => $this->getUser()->getIdString()));
+            if (is_array($mongoResult) && isset($mongoResult['n']))
+                $itemsDeleted = intval($mongoResult['n']);
+            $this->dm()->flush();
+        } catch (Exception $e) {
+            return new JsonResponse('' . $e, 500);
```

Comment:
Yeah, I think I had that there because it shouldn't throw an exception there as far as I am aware, a error 500 would only happen if something unexpected broke or some edge case I haven't thought of.